SonarQube.MSBuild.Tasks/Targets/SonarQube.Integration.targets

```diff
@@ -502,11 +477,12 @@
     </PropertyGroup>
 
     <ItemGroup Condition=" $(SonarLintFound) == 'true' ">
-      <!-- Remove any existing references to files called *SonarLint*.dll -->
-      <Analyzer Remove="@(Analyzer)" Condition="$([System.Text.RegularExpressions.Regex]::IsMatch($([System.String]::new('%(Analyzer.Identity)')), 'SonarLint[^\\]*\.dll$', System.Text.RegularExpressions.RegexOptions.IgnoreCase)) " />
-      <Analyzer Include="$(SonarLintAnalyzerDir)*SonarLint*.dll" />
-      
-      <!-- Add SonarLint.xml as a Roslyn Additional File so that SonarLint can pick up the right rule parameter values -->
+      <!-- Remove any existing analyzers and additional files -->
+      <Analyzer Remove="@(Analyzer)" />
+      <AdditionalFiles Remove="@(AdditionalFiles)" />
+
+      <!-- Add SonarLint analyzers and additional files -->
+      <Analyzer Include="$(SonarLintAnalyzerDir)*SonarLint*.dll" />     
```

Comment:
Simplified logic in the targets: overwrite the Analyzer and AdditionalFiles item groups, and don't merge the rulesets.